components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -1241,6 +1277,39 @@ subroutine shoc_tend_e3sm( &
 
 #endif    
     return         
-  end subroutine shoc_tend_e3sm      
+  end subroutine shoc_tend_e3sm   
+  
+  subroutine grid_size(state, grid_dx, grid_dy)
+  ! Determine the size of the grid for each of the columns in state
+
+  use phys_grid,       only: get_area_p
+  use shr_const_mod,   only: shr_const_pi
+  use physics_types,   only: physics_state
+  use ppgrid,          only: pver, pverp, pcols
+ 
+  type(physics_state), intent(in) :: state
+  real(r8), intent(out)           :: grid_dx(pcols), grid_dy(pcols)   ! E3SM grid [m]
+
+  real(r8), parameter :: earth_ellipsoid1 = 111132.92_r8 ! first coefficient, meters per degree longitude at equator
+  real(r8), parameter :: earth_ellipsoid2 = 559.82_r8 ! second expansion coefficient for WGS84 ellipsoid
+  real(r8), parameter :: earth_ellipsoid3 = 1.175_r8 ! third expansion coefficient for WGS84 ellipsoid
+
+  real(r8) :: mpdeglat, column_area, degree
+  integer  :: i
+
+  ! determine the column area in radians
```

Comment:
Is it correct to say this block of code finds area in radians? It looks like it is finding dx and dy in meters. Maybe this comment applies to the first function call inside the loop?